components/lib/dataprotect/src/main/java/mozilla/components/lib/dataprotect/Keystore.kt

```diff
@@ -235,6 +236,7 @@ open class Keystore(
      * @throws KeystoreException If the data could not be decrypted
      */
     @Throws(KeystoreException::class)
+    @Synchronized
```

Comment:
this could be more granular - just around `val cipher =... \r return cipher.doFinal...`. E.g. you just need to guard access to an instance returned to you by `createDecryptCipher`, other operations don't matter here - and synchronizing over them introduces additional overhead.